packages/web-app-files/src/mixins/spaces/actions/restore.js

```diff
@@ -0,0 +1,72 @@
+import { mapActions } from 'vuex'
+
+export default {
+  computed: {
+    $_restore_items() {
+      return [
+        {
+          name: 'restore',
+          icon: 'restart',
+          label: () => {
+            return this.$gettext('Restore')
+          },
+          handler: this.$_restore_showModal,
+          isEnabled: (space) => {
+            return space.root?.deleted?.state === 'trashed'
+          },
+          componentType: 'oc-button',
+          class: 'oc-files-actions-restore-trigger'
+        }
+      ]
+    }
+  },
+  methods: {
+    ...mapActions([
+      'createModal',
+      'hideModal',
+      'setModalInputErrorMessage',
+      'showMessage',
+      'toggleModalConfirmButton'
+    ]),
+
+    $_restore_showModal(space) {
```

Comment:
Please rename to trigger naming scheme and use array of resources instead of a single resource.